### ISSUE TITLE
[BUG] Document.apply_json_patch losing events

### ISSUE THREAD

**Issue by philippjfr** [bokeh/bokeh#9549]
So this bug report is going to be somewhat vague until I can narrow down exactly what is going on. Basically I am batching events to be sent to Javascript, in my example I have a `FileTransfer` widget which has two properties, `data` and `filename`, both of these properties are updated and sent together. All that happens fine but then something very weird happens when the events arrive on `Document.apply_json_patch`, if I log `events_json` before the first event is applied here: https://github.com/bokeh/bokeh/blob/master/bokehjs/src/lib/document/document.ts#L685, both events show up:

```
(2) [{…}, {…}]
```

however if I log `events_json` immediately after that line only the first event shows up anymore:

```
[{…}]
```

In other words somehow when `patched_obj.setv` is run the second event is dropped and therefore never gets applied. I absolutely cannot explain this and wanted to check whether someone (e.g. @mattpap) has any intuition about what could possible be going on there. `setv` has no reference to `events_json` so I am totally perplexed by this.

**Comment by philippjfr** [comment]
Even more perplexingly the events_json object shows up like this in the console:

![Screen Shot 2019-12-30 at 3 39 43 PM](https://user-images.githubusercontent.com/1550771/71586493-b4c25480-2b1a-11ea-8ceb-9dbe880ffdf0.png)

So it appears like a length two array but if you inspect it, it actually isn't? Another weirdness is that if I push the events into another array the first time they are iterated over and then iterate over that array in the second loop everything works fine.

**Comment by philippjfr** [comment]
Can't reproduce using the usual bokeh machinery so I suspect it's something to do with our setup.
